Library/Homebrew/os/mac/xcode.rb

```diff
@@ -139,7 +139,8 @@ def uncached_version
       end
 
       def provides_autotools?
-        version < "4.3"
+        # Xcode 2.5's autotools are too old to rely on at this point
+        (version < "4.3") && (version > "2.5")
```

Comment:
Grepping the source, it looks like this actually does almost nothing these days - it's a leftover from when we still had the symbol deps. The method and the single place that uses it (to, presumably, no effect) never got removed.